src/plugins/ResizeReorder/ReorderCell.js

```diff
@@ -121,25 +129,139 @@ class ReorderCell extends React.PureComponent {
-        <ReorderHandle
-          {...this.props}
-          translateCell={this.translateCell}
-          onColumnReorderStart={this.onColumnReorderStart}
-          touchEnabled={this.props.touchEnabled}
-          height={this.props.height}
-          isRTL={this.context.isRTL}
-          columnKey={this.props.columnKey}
-          left={this.props.left}
-          onColumnReorderEndCallback={this.onColumnReorderEnd}
-        />
+      <div className={reorderClasses} style={style} ref={this.cellRef}>
+        {this.renderReorderHandle()}
         {content}
       </div>
     );
   }
+
+  renderReorderHandle() {
+    const style = {
+      height: this.props.height,
+    };
+
+    return (
+      <div
+        className={cx({
+          'fixedDataTableCellLayout/columnReorderContainer': true,
+          'fixedDataTableCellLayout/columnReorderContainer/active': false,
+        })}
+        onMouseDown={this.onMouseDown}
+        onTouchStart={this.onTouchStart}
+        style={style}
+      />
+    );
+  }
+
+  onTouchStart = (ev) => {
+    if (!this.props.touchEnabled) {
+      return;
+    }
+
+    this.onMouseDown(ev);
+  };
+
+  /**
+   *
+   * @param {MouseEvent} event
+   */
+  onMouseDown = (event) => {
+    this.onDragStart(event);
+  };
+
+  /**
+   * @param {string} columnKey
+   */
+  onDragStart = (event) => {
+    this.createDragContainer();
+    this.renderDragProxy(event);
+    this.props.onColumnReorderStart(this.props.columnKey);
+  };
+
+  /**
+   * @param ev
+   * @param {string} ev.columnBefore
+   * @param {string} ev.columnAfter
+   * @param {string} ev.reorderColumn
+   */
+  onColumnReorderEnd = (ev) => {
+    if (this.isMounted) {
+      this.setState({
+        isReordering: false,
+      });
+    }
+
+    this.removeDragContainer();
+    this.props.onColumnReorderEnd(ev);
+  };
+
+  /**
+   * Render a proxy version of our cell that can be dragged around without repercussions to component unmounts.
+   */
+  renderDragProxy(reorderStartEvent) {
+    const additionalProps = {
+      isDragProxy: true,
+      reorderStartEvent: reorderStartEvent,
+      onColumnReorderEnd: this.onColumnReorderEnd,
+      contents: this.cellRef.current,
+    };
+
+    ReactDOM.render(
+      // Since we're effectively rendering the proxy in a separate VDOM root, we cannot directly pass in our context.
+      // To solve this, we use ExternalContextProvider to pass down the context value.
+      // ExternalContextProvider also ensures that even if our cell gets unmounted, the dragged cell still receives updates from context.
+      <ExternalContextProvider value={this.context}>
+        <DragProxy {...this.props} {...additionalProps} />
+      </ExternalContextProvider>,
+      this.getDragContainer(),
+      // we consider our cell in a reordering state as soon as the drag proxy gets mounted
+      () => this.setState({ isReordering: true })
+    );
+  }
+
+  createDragContainer = () => {
```

Comment:
Here and below: These are helper methods to manage the drag container which is used as a mount point for the DragProxy.